src/kouta_indeksoija_service/indexer/kouta/common.clj

```diff
@@ -151,14 +151,25 @@
     (postwalk #(-> %
                    (format-date-kws [:koulutuksenAlkamispaivamaara :koulutuksenPaattymispaivamaara :liitteidenToimitusaika :toimitusaika :modified :paattyy :alkaa])) form)))
 
+(defn clean-langs-not-in-kielivalinta [form]
+  (let [langs #{:fi :sv :en}
+        used-langs (set (map #(keyword %) (:kielivalinta form)))
+        langs-to-clean (apply disj langs used-langs)
+        dissoc-langs (fn [x]
+                       (if (map? x) (apply dissoc x langs-to-clean) x))]
+    (if (< 0 (count used-langs))
+      (clojure.walk/postwalk dissoc-langs form)
+      form)))
```

Comment:
Kannattaiskohan tässä vertailla langs-to-cleaniin used-langsin sijaan? Eli ei tartte postwalkata jos kaikki kolme kieltä käytössä